src/services/config_extraction.rs

```diff
@@ -189,7 +192,11 @@ impl ConfigExtractionService for ConfigExtractionServiceImpl {
             require_multiversion_setup: evg_config_utils
                 .get_task_tags(task_def)
                 .contains(MULTIVERSION),
-            multiversion_generate_tasks: evg_config_utils.get_multiversion_generate_tasks(task_def),
+            multiversion_generate_tasks: self
+                .multiversion_service
+                .filter_multiversion_generate_tasks(
+                    evg_config_utils.get_multiversion_generate_tasks(task_def),
+                ),
```

Comment:
Use the `multiversion_service` to filter `multiversion_generate_tasks` so that only the relevant tasks are returned.